docs: add security documentation (#183)
canonical/sqlair: docs/explanation/security.md

# Security

A software layer between your application and database, such as SQLair, can be a
potential security threat. Issues with SQLair could lead to exploits such as SQL
injection or Denial-of-Service through panics or high CPU usage. SQLair
takes a variety of measures to protect users against these attack vectors.

## Argument sanitization
The contents of query arguments extracted by SQLair are passed as parameters to
the database rather than injected into the query as strings. 

> 

Reviewer: So what does this mean for security? Anything the user should do?

Author: Added a sentence about how this mitigates against SQL injection. Theres nothing for the user to do here.

## Struct tag sanitization
Struct tags which are injected into the query as column names are validated

> ensuring that they only contain alphanumeric characters and do not contain

Reviewer: ```suggestion
to ensure that they only contain alphanumeric characters and do not contain
```

> spaces to mitigate against SQL injection.

Reviewer: ```suggestion
spaces. This mitigates against SQL injection.
```

## Query processing
In queries passed to SQLair, SQL not part of a SQLair expressions is not changed

> by SQLair, this mitigates against attacks involving SQLair that touch parts of

Reviewer: ```suggestion
by SQLair. This mitigates against attacks involving SQLair that touch parts of
```
